source/plugin/ray-tracing.rst

```diff
@@ -0,0 +1,80 @@
+===========
+Ray Tracing
+===========
+
+Generically, ray tracing is a method of determining the path of a particle through a coordinate system. In SpongeAPI,
+this is implemented with the ``org.spongepowered.api.util.blockray.BlockRay`` class to discover blocks in the path of
+an arbitrary line in space. A common use case could be to find the block that a Player is looking at.
+
+You can specify the origin of the ray using the ``BlockRay#from`` method, passing in either a ``Location``, an ``Extent``
+and a ``Vector3d``, or an ``Entity``. The method will return a ``BlockRayBuilder``
+
+.. tip::
+    If you specify the origin of the ray to be an ``Entity``, the default direction will be the direction in which the
+    ``Entity`` is pointing.
+
+To specify the end point, you can use the ``BlockRayBuilder#to(Vector3d)`` method, which will set both the direction and
+ending location. Alternatively, you can specify a direction using ``BlockRayBuilder#direction(Vector3d)`` and also a
+block limit using ``BlockRayBuilder#blockLimit(int)``.
+
+.. note::
+    The default block limit is 1000 blocks as a safeguard to prevent infinite iteration. To disable the block limit,
+    use a negative value with ``BlockRayBuilder#blockLimit(int)``.
+
+Filtering
+=========
+Filters determine what blocks are accepted by the ``BlockRay``. To add a filter, use the ``BlockRayBuilder#filter``
+method, passing in one or many ``Predicate<BlockRayHit<E>>``\ s (where ``E`` extends ``Extent``). For convenience,
+``BlockRay`` contains the following methods for common filter use cases:
+
+ * ``allFilter``: returns a filter accepting all blocks
+ * ``onlyAirFilter``: returns a filter accepting only air
+ * ``blockTypeFilter(BlockType)``: returns a filter accepting only the specified ``BlockType``
+ * ``maxDistanceFilter(Vector3d, double)``: returns a filter that stops at certain distance from the given ``Vector3d``
+ * ``continueAfterFilter(Predicate<BlockRayHit<E>>, int)``: returns a filter that continues past the given filter by
+   the specified number of blocks.
+
+Of course, you can also write your own ``Predicate<BlockRayHit<E>>`` filter.
+
+Finally, use ``BlockRayBuilder#build`` to finish building the ``BlockRay``. An example usage of ``BlockRayBuilder`` to
+get the first non-air block a ``Player`` is looking at is below.
```

Comment:
I think this needs a word between "is" and "below". I suggest shown, given or provided.